vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/WebClientOauth2Aware.java

```diff
@@ -0,0 +1,78 @@
+package io.vertx.ext.web.client.impl;
+
+import io.vertx.ext.auth.User;
+import io.vertx.ext.auth.authentication.Credentials;
+import io.vertx.ext.auth.oauth2.OAuth2Auth;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOAuth2;
+
+public class WebClientOauth2Aware extends WebClientBase implements WebClientOAuth2 {
+
+  private final OAuth2Auth oauth2Auth;
+  private Credentials credentials;
+
+  private User user;
+  private boolean withAuthentication;
```

Comment:
This field should be removed as now we don't need explicit call to know if we are going to apply the wrapper logic. We can check instead credentials field.